eslint-bridge/src/linting/eslint/rules/no-clear-text-protocols.ts

```diff
@@ -288,23 +293,124 @@ function checkGroup(expr: estree.NewExpression, ctx: Rule.RuleContext) {
     return;
   }
 
-  function isUnknown(node: estree.Node | undefined, value: estree.Node | undefined) {
-    return node?.type === 'Identifier' && !isUndefined(node) && value === undefined;
+  function report(node: estree.Node) {
+    ctx.report({
+      messageId: 'replicationGroup',
+      node,
+    });
+  }
+}
+
+const awsKinesisRule: Rule.RuleModule = AwsCdkTemplate(
+  {
+    'aws-cdk-lib.aws_kinesis.Stream': checkStream,
+    'aws-cdk-lib.aws_kinesis.CfnStream': checkCfnStream,
+  },
+  {
+    meta: {
+      messages: {
+        encryptionDisabled: 'Make sure that disabling stream encryption is safe here.',
+      },
+    },
+  },
+);
+
+function checkStream(expr: estree.NewExpression, ctx: Rule.RuleContext) {
+  const argument = expr.arguments[2];
+  if (argument == null || isUndefined(argument)) {
+    return;
+  }
+
+  const props = getValueOfExpression(ctx, argument, 'ObjectExpression');
+
+  if (isUnknown(argument, props) || props === undefined) {
+    return;
+  }
+
+  const encryption = getProperty(props, ENCRYPTION, ctx);
+  if (!encryption) {
+    return;
+  }
+
+  const encryptionValue = getUniqueWriteUsageOrNode(ctx, encryption.value);
+
+  if (isUnencryptedStream(encryptionValue)) {
+    report(encryption.value);
+    return;
+  }
+
+  function isUnencryptedStream(node: estree.Node) {
+    if (!isMemberIdentifier(node)) {
+      return false;
+    }
+    const className =
+      node.object.type === 'Identifier' ? node.object.name : node.object.property.name;
+    const constantName = node.property.name;
+    return className === 'StreamEncryption' && constantName === 'UNENCRYPTED';
```

Comment:
I think you can reuse the `getFullyQualifiedName` helper, which is more reliable, concise and handles cases where `StreamEncryption.UNENCRYPTED` is assigned to a variable first.

```suggestion
  function isUnencryptedStream(node: estree.Node) {
    return getFullyQualifiedName(ctx, node)?.replace(/-/g, '_') === 'aws_cdk_lib.aws_kinesis.StreamEncryption.UNENCRYPTED'
```

Reply:
Done